### PR TITLE
Implement HCNNetworkWatcher

### PR DESCRIPTION
**Pull request by jagadishmurugan**

### CODE REVIEW

File: hcn/hcnnetwork.go [hunk 1]
```diff
@@ -123,6 +123,14 @@ type HcnServiceWatcher struct {
 	started        bool
 }
 
+type HcnNetworkWatcher struct {
+	handleLock     sync.RWMutex
+	network        hcnNetwork
```
Comment (jagadishmurugan) [inline hcn/hcnnetwork.go]: remove

File: hcn/hcnnetwork.go [hunk 2]
```diff
@@ -213,6 +221,104 @@ func (w *HcnServiceWatcher) Close() error {
...
+	w.handleLock.RLock()
+	defer w.handleLock.RUnlock()
+
+	if w.started {
+		return nil
+	}
+
+	var networkHandle hcnNetwork
+	var resultBuffer *uint16
+	networkGuid, err := guid.FromString(networkId)
```
Comment (johnsonshih) [inline hcn/hcnnetwork.go:250]: >err [](http://example.com/codeflow?start=14&length=3)

check if err occurs.

Or consider pass guid type for networkId (if applicable)
Or pass networkHandle??? #Closed

Comment (jagadishmurugan) [reply hcn/hcnnetwork.go:250]: We dont need either, we can return errInvalidNetworkID like the error is handled in existing impl...

**Review by johnsonshih — APPROVED** [review]
:shipit:

**Pull request merged**